modules/strelka/1.0/strelka.smk

```diff
@@ -0,0 +1,213 @@
+        """)
+
+rule _strelka_configure_unpaired: # germline
+    input:
+        tumour_bam = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/{tumour_id}.bam",
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa")
+    output:
+        runwf = CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/runWorkflow.py"
+    log:
+        stdout = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_configure.stdout.log",
+        stderr = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_configure.stderr.log"
+    params:
+        opts = op.switch_on_wildcard("seq_type", CFG["options"]["configure"]),
+    wildcard_constraints:
+        pair_status = "no_normal"
+    conda:
+        CFG["conda_envs"]["strelka"]
+    shell:
+        op.as_one_line("""
+        configureStrelkaGermlineWorkflow.py 
+        --bam={input.tumour_bam}
+        --referenceFasta={input.fasta}
+        --runDir=$(dirname {output.runwf})
+        {params.opts} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+
+# Example variant filtering rule (single-threaded; can be run on cluster head node)
+rule _strelka_run:
+    input:
+        runwf = CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/runWorkflow.py"
+    output:
+        vcf_dir = directory(CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/results/variants/")
+        #snp = CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/results/variants/somatic.snvs.vcf.gz",
+        #indels = CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/results/variants/somatic.indels.vcf.gz"
+    log:
+        stdout = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_run.stdout.log",
+        stderr = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_run.stderr.log"
+    params:
+        opts = CFG["options"]["strelka"]
+    conda:
+        CFG["conda_envs"]["strelka"]
+    threads:
+        CFG["threads"]["strelka"]
+    resources: 
+        mem_mb = CFG["mem_mb"]["strelka"]
+    shell:
+        op.as_one_line("""
+        {input.runwf} -j {threads} 
+        {params.opts} 
+        >{log.stdout} 2> {log.stderr}
+        """)
+
+
+rule _strelka_unzip:
+    input:
+        vcf_dir = rules._strelka_run.output.vcf_dir
+        #vcf = CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/results/variants/{var_type}.vcf.gz"
+    output:
+        vcf = CFG["dirs"]["filtered"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{var_type}.vcf"
+    shell:
+        "zcat {input.vcf_dir}/{wildcards.var_type}.vcf.gz > {output.vcf}"
```

Comment:
Let's rely on `gzip -dc` instead of `zcat` for consistency. I don't know how standard `zcat` is.